File: 250504/색종이의 총 넓이/the-total-area-of-colored-paper.cpp
#include <iostream>

#define MAX_N 201
#define OFFSET 100 // 좌표 양수
using namespace std;

int n;
int x, y;
bool area[MAX_N][MAX_N]; // 직사각형 면적
int main()
{
    ios::sync_with_stdio(0);
    cin.tie(NULL);
    cout.tie(NULL);

    cin >> n;
    for(int i = 0; i < n; i++) {
        cin >> x >> y;
        for(int j = x + OFFSET; j < x + 8 + OFFSET; j++) {
            for(int k = y + OFFSET; k < y + 8 + OFFSET; k++) {
                area[j][k] = true; // 직사각형 면적
            }
        }
    }
    int cnt = 0;
    for(int i = 0; i < MAX_N; i++) {
        for(int j = 0; j < MAX_N; j++) {
            if(area[i][j]) {
                cnt++;
            }
        }
    }
    cout << cnt << "\n"; // 면적의 넓이 출력
    return 0;
}

File: 250504/색종이의 총 넓이/the-total-area-of-colored-paper.java
import java.util.Scanner;

public class Main {

    public static final int MAX_N = 201;
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        boolean[][] area = new boolean[MAX_N][MAX_N];
        int offset = 100;
        int n = sc.nextInt();

        for(int i = 0; i < n; i++) {
            int x = sc.nextInt() + offset;
            int y = sc.nextInt() + offset;
            for(int j = x; j < x + 8; j++) {
                for(int k = y; k < y + 8; k++) {
                    area[j][k] = true;
                }
            }
        }
        int count = 0;
        for(int i = 0; i < MAX_N; i++) {
            for(int j = 0; j < MAX_N; j++) {
                if(area[i][j]) {
                    count++;
                }
            }
        }
        System.out.println(count);
        sc.close();
    }
}
